### DISCUSSION TITLE
cpp_id_attributes and 'function like' macros?

### DISCUSSION THREAD

**Discussion by blitz-research** [michaeljones/breathe#997] [General]
Hi,

Is cpp_id_attributes supposed to work with 'function like' macros, eg: for something like:
```
SGD_FLAGS(SGD_EventMask) enum SGD_EventMask {
};
```
...and in my conf.py:
```
cpp_id_attributes = ["SGD_FLAGS"]
```
This is currently get ommited from the docs unless I remove SGD_FLAGS.

I can add SGD_FLAGS to Doxfile PREDEFINED and the enum gets docced again, but this removes SGD_FLAGS from the docs and I'd quite like to have it in there.

Bye,
Mark

**Comment by jakobandersen** [comment]
No, but there is ``cpp_paren_attributes`` (https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-cpp_paren_attributes).

**Reply by blitz-research** [reply]
D'oh, thanks!
